tests/e2e/asserts_test.go

```diff
@@ -2426,3 +2426,59 @@ func AssertPvcHasLabels(
 		}, 300, 5).Should(Succeed())
 	})
 }
+
+// AssertRepSlotsOnPod checks if all the required replication slot exists in a given pod.
+// In case we are targeting the primary, it will also check if the slot is active.
+func AssertRepSlotsOnPod(
+	namespace,
+	clusterName,
+	podName string,
+) {
+	namespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      podName,
+	}
+	targetPod := &corev1.Pod{}
+	err := env.Client.Get(env.Ctx, namespacedName, targetPod)
+	Expect(err).ToNot(HaveOccurred())
+
+	slots, err := testsUtils.DiscoverRepSlotsOnPod(namespace, clusterName, podName, env)
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, slot := range slots {
```

Comment:
Already verified all expected replication slot is in this pod, vice versa, going to add logic to verify all existing replication slot is as expected, no redundant. this could be useful. when scale down or switch over, check redundant replication shot is deleted.

Reply:
> going to add logic to verify all existing replication slot is as expected, no redundant. this could be useful. when scale down or switch over, check redundant replication shot is deleted.

We already have a list of expected rep_slots here, so we could gather the current list of rep_slots on the targetPod and expect the 2 lists to eventually be equivalent (that would cover also scale down/switchover where "old" slots needs to be removed by the operator)

Reply:
Done.